src/mat.c

```diff
@@ -1691,15 +1691,17 @@ Mat_VarGetSize(matvar_t *matvar)
     int i;
     size_t bytes = 0, overhead = 0, ptr = 0;
 
-#if defined(_WIN64) || (defined(__SIZEOF_POINTER__) && (__SIZEOF_POINTER__ == 8)) || (defined(SIZEOF_VOID_P) && (SIZEOF_VOID_P == 8))
-    /* 112 bytes cell/struct overhead for 64-bit system */
-    overhead = 112;
-    ptr = 8;
-#elif defined(_WIN32) || (defined(__SIZEOF_POINTER__) && (__SIZEOF_POINTER__ == 4)) || (defined(SIZEOF_VOID_P) && (SIZEOF_VOID_P == 4))
-    /* 60 bytes cell/struct overhead for 32-bit system */
-    overhead = 60;
-    ptr = 4;
-#endif
+    if (sizeof(void*)==8) {
+        /* 112 bytes cell/struct overhead for 64-bit system */
+        overhead = 112;
+        ptr = 8;
+    } else if (sizeof(void*)==4) {
+        /* 60 bytes cell/struct overhead for 32-bit system */
+        overhead = 60;
+        ptr = 4;
+    } else {
+        Matio_Error("Pointers of size %d are not supported by matio.",sizeof(void*)*CHAR_BIT);
+    }
```

Comment:
This changes a compiletime check by a run-time check. I'd rather have it as it is as it already uses compiler defines used by MSVC or GCC/clang.